cvat/apps/annotation/README.md

```diff
@@ -150,3 +150,352 @@ It allows to download and upload annotations in different formats and easily add
+   ```bash
+   aeroplane
+   bicycle
+   bird
+   boat
+   bottle
+   bus
+   car
+   cat
+   chair
+   cow
+   diningtable
+   dog
+   horse
+   motorbike
+   person
+   pottedplant
+   sheep
+   sofa
+   train
+   tvmonitor
+   ```
+1. Zip all label files together (we need to add only label files that correspond to the train subset)
+   ```bash
+   cat train.txt | while read p; do echo ${p%/*/*}/labels/${${p##*/}%%.*}.txt; done | zip labels.zip -j -@ obj.names
+   ```
+1. Click `Upload annotation` button, choose `YOLO ZIP 1.0` and select the *.zip file with labels from previous step.
+   It may take some time.
+
+### [MS COCO Object Detection](http://cocodataset.org/#format-data)
+#### COCO dumper description
+- downloaded file: single unpacked `json`. Detailed description of the MS COCO format can be found [here](http://cocodataset.org/#format-data)
+- supported shapes - Polygons, Rectangles (interpreted as polygons)
+
+#### COCO loader description
+- uploaded file: single unpacked `*.json`.
+- supported shapes: Polygons (the `segmentation` must not be empty)
+- additional comments: the CVAT task should be created with the full label set that may be in the annotation files
+
+#### How to create a task from MS COCO dataset
+1.  Download the [MS COCO dataset](http://cocodataset.org/#download).
+    For example [2017 Val images](http://images.cocodataset.org/zips/val2017.zip)
+    and [2017 Train/Val annotations](http://images.cocodataset.org/annotations/annotations_trainval2017.zip).
+1.  Create a CVAT task with the following labels:
+      ```bash
+      person bicycle car motorcycle airplane bus train truck boat "traffic light" "fire hydrant" "stop sign" "parking meter" bench bird cat dog horse sheep cow elephant bear zebra giraffe backpack umbrella handbag tie suitcase frisbee skis snowboard "sports ball" kite "baseball bat" "baseball glove" skateboard surfboard "tennis racket" bottle "wine glass" cup fork knife spoon bowl banana apple sandwich orange broccoli carrot "hot dog" pizza donut cake chair couch "potted plant" bed "dining table" toilet tv laptop mouse remote keyboard "cell phone" microwave oven toaster sink refrigerator book clock vase scissors "teddy bear" "hair drier" toothbrush
+      ```
+
+      Select val2017.zip as data
+      (See [Creating an annotation task](cvat/apps/documentation/user_guide.md#creating-an-annotation-task)
+      guide for details)
+1.  unpack annotations_trainval2017.zip
+1.  click `Upload annotation` button,
+    choose `COCO JSON 1.0` and select `instances_val2017.json.json` annotation file. It may take some time.
+
+### [TFRecord](https://www.tensorflow.org/tutorials/load_data/tf_records)
+TFRecord is a very flexible format, but we try to correspond the format that used in
+[TF object detection](https://github.com/tensorflow/models/tree/master/research/object_detection)
+with minimal modifications.
+Used feature description:
+```python
+image_feature_description = {
```

Comment:
I saw a couple of fixes in the code regrading types. Could you please confirm that the definition is correct?

Reply:
It's up to date